### PR TITLE
[BE] - remove JS build, demo, and related handling

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

This PR removes the JS build and bindings from habitat-sim. 

Context: Given the paradigm shift to habitat-lab's Human in the Loop (HitL) client-server architecture, it no longer makes sense to maintain and develop direct web deployment of habitat-sim on web through JS.

## How Has This Been Tested

CI should be updated to no longer expect any JS build, test, or linting.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -14,18 +14,14 @@ commands:
...
               cat ./hablab_sha
               wget https://cmake.org/files/v3.12/cmake-3.12.4-Linux-x86_64.sh
-              wget https://cmake.org/files/v3.20/cmake-3.20.1-linux-x86_64.sh
               sudo mkdir /opt/cmake312
-              sudo mkdir /opt/cmake320
               sudo sh ./cmake-3.12.4-Linux-x86_64.sh --prefix=/opt/cmake312 --skip-license
-              sudo sh ./cmake-3.20.1-linux-x86_64.sh --prefix=/opt/cmake320 --skip-license
               sudo ln -s /opt/cmake312/bin/cmake /usr/local/bin/cmake
       - run: &install_deps
           name: Install dependencies
```
Comment (eundersander) [inline .circleci/config.yml:27]: What about "Install Headless Chrome dependencies" below? Maybe that is only for testing web stuff?

**Review by eundersander — COMMENTED** [review]
How about also search the codebase for EMSCRIPTEN? For example this block in viewer's CMakeLists.txt:
```
# copy JS/HTML/CSS scaffolding for WebGL build
if(CORRADE_TARGET_EMSCRIPTEN)
  install(FILES viewer.html ${MAGNUM_EMSCRIPTENAPPLICATION_JS}
                ${MAGNUM_WEBAPPLICATION_CSS} DESTINATION ${CMAKE_CURRENT_BINARY_DIR}
  )
endif()
```

**Comment by aclegg3** [comment]
> How about also search the codebase for EMSCRIPTEN? For example this block in viewer's CMakeLists.txt (UPDATE: ahh I guess you already found this one):

Good suggestion, found a bunch more of these pivots. I agree with not removing things from the magnum cmake files. I did remove these from our base cmake and dependency files.

**Review by eundersander — APPROVED** [review]
LGTM!

**Comment by aclegg3** [comment]
> LGTM! Edit: actually I'm still curious about "Install Headless Chrome dependencies" (see my comment), but otherwise LGTM!

Yeah, I wanted to see a passing test before removing that to get a clear signal about the effect of removal. Will try that before merging, thanks!

**Pull request merged**